src/main/java/com/example/demo/controller/CarController.java

```diff
@@ -0,0 +1,125 @@
+package com.example.demo.controller;
+
+import com.example.demo.model.*;
+import com.example.demo.repository.CarRepository;
+import com.example.demo.repository.RentRepository;
+import com.example.demo.service.UserService;
+import com.example.demo.validator.RentValidator;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+
+import javax.persistence.criteria.*;
+import java.util.ArrayList;
+import java.util.List;
+
+@RestController
+public class CarController {
+
+    @Autowired
+    private CarRepository carRepository;
+
+    @Autowired
+    private RentRepository rentRepository;
+
+    @Autowired
+    private RentValidator rentValidator;
+
+    @Autowired
+    private UserService userService;
+
+
+
+    @GetMapping("/rent")
+    public String rent(Model model) {
+        model.addAttribute("rentForm", new Rent());
+        return "registration";
+    }
+
+    // do tego post muszę dostać rentForm z Car i User już
+    @PostMapping("/rent")
+    public List<String> rent(@RequestBody Rent rentForm) {
+
+        //to jakbym nie dostała User z frontu
+        /*
+        Object principal = SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        String email = ((UserDetails)principal).getUsername();
+        User user = userService.findByEmail(email);
+         */
+
+        List<String> messages = rentValidator.validate(rentForm);
+        if(messages.isEmpty()) {
+            rentRepository.save(rentForm);
+        }
+        return messages;
```

Comment:
Powinniśmy zwracać JSON czyli string opakowany w obiekcie ponieważ łatwiej jest wtedy to odczytać w froncie oraz unika się wtedy jakiś blędów

Reply:
to samo tyczy się errorów, zwracamy obiekt z polami message oraz error, zwróćmy uwage żeby message było ładnie napisane bo wtedy front sobie to ładnie wyświetli na stronie